fcrepo-client-impl/src/main/java/org/fcrepo/client/impl/FedoraRepositoryImpl.java

```diff
@@ -220,6 +221,33 @@ public FedoraObject createObject(final String path) throws FedoraException {
     }
 
     @Override
+    public FedoraObject createResource(final String containerPath) throws FedoraException {
+        final HttpPost post = httpHelper.createPostMethod(containerPath == null ? "" : containerPath, null);
+        try {
+            final HttpResponse response = httpHelper.execute(post);
+            final String uri = post.getURI().toString();
+            final StatusLine status = response.getStatusLine();
+            final int statusCode = status.getStatusCode();
+
+            if (statusCode == SC_CREATED) {
+                return getObject(response.getFirstHeader("Location").getValue().substring(repositoryURL.length()));
+            } else if (statusCode == SC_FORBIDDEN) {
+                LOGGER.error("request to create resource {} is not authorized.", uri);
+                throw new ForbiddenException("request to create resource " + uri + " is not authorized.");
+            } else {
+                LOGGER.error("error creating resource {}: {} {}", uri, statusCode, status.getReasonPhrase());
+                throw new FedoraException("error retrieving resource " + uri + ": " + statusCode + " " +
```

Comment:
Mismatch between LOGGER output ("creating") and exception message ("retrieving").